version/version.go

```diff
@@ -20,4 +20,45 @@
 
 package version
 
+import "log"
+
+// Version is the current thriftrw version.
 const Version = "0.4.0"
+
+// CheckCompatibilityWithGeneratedCodeAt will panic if the thriftrw version
+// used to generated code (given by `genCodeVer`) is not compatible with the
+// current version of thriftrw.
+// This function is designed to be called during initialization of the
+// generated code.
+//
+// Rational: Let's say you use thriftrw version 1.0 to generate some stubs.
+// Later on, you imports the stubs, but also thriftrw in version 1.2. Maybe
+// thriftrw 1.2 is not compatible in subtle ways with the generated code from
+// version 1.0. This function will make sure to panic during initialization
+// preventing potential bugs.
+func CheckCompatibilityWithGeneratedCodeAt(genCodeVersion string, fromPkg string) {
+	genv := parseSemVerOrPanic(genCodeVersion)
+	compatible := (genv.Compare(&genCodeCompatbilityRange.begin) >= 0 &&
+		genv.Compare(&genCodeCompatbilityRange.end) < 0)
+	if !compatible {
+		log.Panicf(`incompatible version from generaged package "%s", expected >=%s and <%s, got %s`,
+			fromPkg, &genCodeCompatbilityRange.begin,
+			&genCodeCompatbilityRange.end, &genv)
+	}
+}
+
+var genCodeCompatbilityRange = computeGenCodeCompabilityRange()
+
+type genCodeCompatbilityRangeHolder struct {
```

Comment:
I'm not really concerned about the `init`. I just think the type is
unnecessary and an anonymous struct will suffice. The reason to do it in
`init` is because returning the anonymous type from a function will mean
repeating it.

Also, correction about the anonymous struct example: You would actually just
do:

```
var compatibleRange struct {
    Begin *semVer
    End *semVer
}

func init() {
    compatibleRange.Begin = ...
    compatibleRange.End = ...
}
```